src/TrackingOptOut.php

```diff
@@ -0,0 +1,141 @@
+<?php
+
+/**
+ * @file
+ * Contains \Netzstrategen\CoreStandards\TrackingOptOut.
+ */
+
+namespace Netzstrategen\CoreStandards;
+
+class TrackingOptOut {
+
+  /**
+   * @implements admin_init
+   */
+  public static function admin_init() {
+    static::addSettingsFields();
+  }
+
+  /**
+   * @implements init
+   */
+  public static function init() {
+
+    // Register the shortcode to call the opt-out cookie creation script.
+    add_shortcode('opt-out-link', __CLASS__ . '::registerOptOutShortcode');
+
+    // Enqueue frontend scripts.
+    if (get_option('core_standards_opt_out_active')) {
+      add_action('wp_enqueue_scripts', __CLASS__ . '::wp_enqueue_scripts');
+      // Injects the Google Analytics IDs and the opt-out script in the page head
+      // section before any tracking script.
+      add_action('wp_head', __CLASS__ . '::wp_head', 5);
+      add_action('wp_enqueue_scripts', __CLASS__ . '::enqueueScriptsFirstHead', 1);
+    }
+  }
+
+  /**
+   * Adds Opt-Out settings fields to 'Reading' settings page.
+   */
+  public static function addSettingsFields() {
+    register_setting(
+      'reading',
+      'core_standards_opt_out_ids',
+      'esc_attr'
+    );
+    add_settings_field(
+      'core_standards_opt_out_ids',
+      __('Google Analytics IDs', Plugin::L10N),
+      __CLASS__ . '::optOutGoogleIdsFieldRender',
+      'reading',
+      'default'
+    );
+
+    register_setting(
+      'reading',
+      'core_standards_opt_out_active'
+    );
+    add_settings_field(
+      'core_standards_opt_out_active',
+      __('Activate Opt-Out', Plugin::L10N),
+      __CLASS__ . '::optOutActiveFieldRender',
+      'reading',
+      'default'
+    );
+  }
+
+  /**
+   * Renders Google Analytics IDs list field.
+   */
+  public static function optOutGoogleIdsFieldRender() {
+    echo '<textarea id="core-standards-opt_out-ids" name="core_standards_opt_out_ids" rows="8" cols="40">' .
+      esc_attr(get_option('core_standards_opt_out_ids')) . '</textarea>';
+  }
+
+  /**
+   * Renders Google Analytics Opt-Out activation checkbox field.
+   */
+  public static function optOutActiveFieldRender() {
+    // Deactivates opt-out if Google Analytics IDs fiels is empty.
+    if (!get_option('core_standards_opt_out_ids')) {
+      update_option('core_standards_opt_out_active', FALSE);
```

Comment:
Performing a write operation during a read request is an absolute no-go.  Never ever do this.

If there is a dependency between two options, then you need to adjust the POST values before they are processed by the Settings API.

(Of course, it would be much easier to simply have just one value, but…)